### ISSUE TITLE
can't "create" where upper directory is reachable but non-readable

### ISSUE THREAD

**Issue by dmarck** [borgbackup/borg#7746]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

BUG

#### Your borg version (borg -V).

borg 1.2.0

#### Operating system (distribution) and version.

ubuntu server 22.04

## Describe the problem you're observing.

where parent of backing up tree is not readable (but accessible, like 0711), borg can't start backup with the following exception

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

```
mkdir -m 711 /fs/parent
mkdir -m 755 /fs/parent/tree
touch /fs/parent/tree/backupfile.txt
```

(from other user) 

```
borg create /path/to/repo::tag /fs/parent/tree
```

#### Include any warning/errors/backtraces from the system logs

```
marck@box:~$ borg --debug create --progress --stats borg/repo::Initial /fs/users/marck/backups/games/
...
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/helpers/fs.py", line 290, in os_open
    fd = os.open(fname, _flags_noatime, dir_fd=parent_fd)
PermissionError: [Errno 13] Permission denied: '/fs/users/marck/backups'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archive.py", line 264, in OsOpen
    fd = os_open(path=path, parent_fd=parent_fd, name=name, flags=flags, noatime=noatime)
```

**Comment by ThomasWaldmann** [comment]
Can you reproduce that with 1.2.4 also?

**Comment by ThomasWaldmann** [comment]
Looking at the traceback, guess that is because borg (since 1.2) tries to always work with FDs rather than file names to avoid race conditions.

That is not only done for files, but also for directories, including the directory where the recursion starts.

For that it opens the parent directory and gets its FD and then opens the wanted directory giving its basename and the parent directory FD.

**Comment by ThomasWaldmann** [comment]
I thought about this and although the setup with unreadable parent is a bit unusual, I guess we have to consider this a bug.

**Comment by ThomasWaldmann** [comment]
Will be fixed in 1.2.5 release (soon).
